src/main/java/com/google/sps/ReachableNodesTest.java

```diff
@@ -0,0 +1,258 @@
+package com.google.sps;
+
+import java.util.HashMap;
+import java.util.Set;
+
+import com.google.common.graph.EndpointPair;
+import com.google.common.graph.MutableGraph;
+import com.proto.GraphProtos.Node;
+import com.proto.GraphProtos.Node.Builder;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ReachableNodesTest {
```

Comment:
I think these are all great tests! You could also add a test similar to the one you have in MaxDepthTest.java where a node is a reachable parent/child through one path but not another

Reply:
good idea!